packages/devtools_app/lib/src/inspector/flutter/inspector_screen.dart

```diff
@@ -25,42 +36,223 @@ class InspectorScreen extends Screen {
+    return Align(
+      alignment: Alignment.topRight,
+      child: Container(
+        padding: const EdgeInsets.symmetric(horizontal: 6),
+        // Add a semi-transparent background to the
+        // expand and collapse buttons so they don't interfere
+        // too badly with the tree content when the tree
+        // is narrow.
+        color: Theme.of(context).scaffoldBackgroundColor.withAlpha(200),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.end,
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Flexible(
+              child: OutlineButton(
+                onPressed: enableButtons ? _onExpandClick : null,
+                child: const Text(
+                  'Expand all',
+                  overflow: TextOverflow.ellipsis,
+                ),
+              ),
+            ),
+            Flexible(
+              child: OutlineButton(
+                onPressed: enableButtons ? _onResetClick : null,
+                child: const Text(
+                  'Collapse to selected',
+                  overflow: TextOverflow.ellipsis,
+                ),
+              ),
+            )
+          ],
+        ),
+      ),
+    );
+  }
+
+  void _onExpandCollapseSupported() {
+    setState(() {
+      _expandCollapseSupported = true;
+    });
+  }
+
+  void _handleConnectionStart(VmService service) async {
+    setState(() {
+      connectionInProgress = true;
+    });
+
+    try {
+      // TODO(jacobr): move this rootBundle loading code into
+      // InspectorController once the dart:html app is removed and Flutter
+      // conventions for loading assets can be the default.
+      if (Catalog.instance == null) {
+        final json = await rootBundle.loadString('web/widgets.json');
+        // ignore: invalid_use_of_visible_for_testing_member
+        Catalog.setCatalog(Catalog.decode(json));
+      }
+      await ensureInspectorServiceDependencies();
+
+      // Init the inspector service, or return null.
+      inspectorService =
+          await InspectorService.create(service).catchError((e) => null);
+    } finally {
+      setState(() {
+        connectionInProgress = false;
+      });
+    }
+
+    if (inspectorService == null) {
+      return;
+    }
+
+    setState(() {
+      summaryTreeController = InspectorTreeControllerFlutter();
+      detailsTreeController = InspectorTreeControllerFlutter();
+      inspectorController = InspectorController(
+        inspectorTree: summaryTreeController,
+        detailsTree: detailsTreeController,
+        inspectorService: inspectorService,
+        treeType: FlutterTreeType.widget,
+        onExpandCollapseSupported: _onExpandCollapseSupported,
+      );
+
+      // TODO(jacobr): update visibility based on whether the screen is visible.
+      // That will reduce memory usage on the device running a Flutter
+      // application when the inspector panel is not visible.
+      inspectorController.setVisibleToUser(true);
```

Comment:
good point. done.
added the extra plumbing from the inspectorTreeController to the inspectorController so that the inspectorController deactivates when the last inspectorTreeView deactivates.

Reply:
we are. the dispose method is clearing the client.